programs/steward/src/instructions/close_directed_stake_ticket.rs

```diff
@@ -0,0 +1,48 @@
+use anchor_lang::prelude::*;
+
+use crate::{errors::StewardError, Config, DirectedStakeTicket};
+use std::mem::size_of;
+
+#[derive(Accounts)]
+pub struct CloseDirectedStakeTicket<'info> {
+    #[account()]
+    pub config: AccountLoader<'info, Config>,
+
+    #[account(
+        mut,
+        close = authority
+    )]
+    pub ticket_account: AccountLoader<'info, DirectedStakeTicket>,
```

Comment:
I believe we need to add the config pubkey to the derivation of the ticket_account. Here, and in the respective init instruction for ticket accounts.

It seems that there is a bug here, where someone could pass in a "rogue" config account, and close this ticket account.

The flow would be like this. A random keypair goes to the spl-stake-pool program and creates a stake pool. From what I can tell, that is a permissionless instruction. Then they would go to the steward program and create a new steward account and respective config. That config would specify their own `.directed_stake_whitelist_authority`. And then pass that authority in as a signer here.

I've scanned the steward program, and this ticket account is the only PDA that does not use the config pubkey as a seed.